flink-core/src/main/java/org/apache/flink/util/InstantiationUtil.java

```diff
@@ -298,6 +299,21 @@ public static void writeObjectToConfig(Object o, Configuration config, String ke
 			return baos.toByteArray();
 		}
 	}
+
+	/**
+	 * Clones the given serializable object using Java serialization.
+	 *
+	 * @param obj Object to clone
+	 * @param classLoader class loader to use to deserialize the object
+	 * @param <T> Type of the object to clone
+	 * @return Cloned object
+	 * @throws IOException
+	 * @throws ClassNotFoundException
+	 */
+	public static <T extends Serializable> T clone(T obj, ClassLoader classLoader) throws IOException, ClassNotFoundException {
```

Comment:
I think we can drop the classloader here and simply use the classloader of the original object:

`return deserializeObject(serializedObject, obj.getClass().getClassLoader());`